@flavorjones Thanks, this is awesome! I do see these warnings at the beginning of the build:

```
﻿﻿no configuration section for specified version of Ruby (rbconfig-x86-linux-2.5.0)
no configuration section for specified version of Ruby (rbconfig-x86-linux-2.6.0)
no configuration section for specified version of Ruby (rbconfig-x86-linux-2.7.0)
no configuration section for specified version of Ruby (rbconfig-x86-linux-3.0.0)
```

Is that expected?

After installing the gem, I also got this:

```
irb(main):001:0> require 'magic'
Traceback (most recent call last):
       10: from /Users/stanhu/.rbenv/versions/2.7.2/bin/irb:23:in `<main>'
        9: from /Users/stanhu/.rbenv/versions/2.7.2/bin/irb:23:in `load'
        8: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/irb-1.2.6/exe/irb:11:in `<top (required)>'
        7: from (irb):1
        6: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:156:in `require'
        5: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `rescue in require'
        4: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:168:in `require'
        3: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/ruby-magic-0.4.0-x86_64-linux/lib/magic.rb:3:in `<top (required)>'
        2: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/ruby-magic-0.4.0-x86_64-linux/lib/magic.rb:7:in `rescue in <top (required)>'
        1: from /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/ruby-magic-0.4.0-x86_64-linux/lib/magic.rb:7:in `require_relative'
LoadError (cannot load such file -- /Users/stanhu/.rbenv/versions/2.7.2/lib/ruby/gems/2.7.0/gems/ruby-magic-0.4.0-x86_64-linux/lib/magic/magic)
```

Seems we are missing the `share` files that are normally installed with the `libmagic` compilation.

--- a/Rakefile
+++ b/Rakefile
@@ -17,8 +17,9 @@ CLEAN.include FileList['**/*{.o,.so,.dylib,.bundle}'],
 CLOBBER.include FileList['**/tmp'],
                 FileList['**/*.log'],
                 FileList['doc/**'],
-                FileList['ports/'],
                 FileList['tmp/']
+
+CLOBBER.add("ports/*").exclude(%r{ports/archives$})
 
 RUBY_MAGIC_GEM_SPEC = Gem::Specification.load('ruby-magic.gemspec')
 

--- a/ext/magic/ruby-magic.c
+++ b/ext/magic/ruby-magic.c
@@ -299,10 +299,7 @@ rb_mgc_get_paths(VALUE object)
 		return value;
 
 	cstring = magic_getpath_wrapper();
-	value = magic_split(CSTR2RVAL(cstring), CSTR2RVAL(":"));
-	RB_GC_GUARD(value);
-
-	return value;
+	return magic_set_paths(object, magic_split(CSTR2RVAL(cstring), CSTR2RVAL(":")));
 }
 
 /*
@@ -487,14 +484,11 @@ rb_mgc_load(VALUE object, VALUE arguments)
 
 	ma.flags = magic_get_flags(object);
 
-	if (!RARRAY_EMPTY_P(arguments)) {
-		value = magic_join(arguments, CSTR2RVAL(":"));
-		ma.type.file.path = RVAL2CSTR(value);
+	if (RARRAY_EMPTY_P(arguments)) {
+		arguments = rb_mgc_get_paths(object);
 	}
-	else
-		ma.type.file.path = magic_getpath_wrapper();
-
-	magic_set_paths(object, RARRAY_EMPTY);
+	value = magic_join(arguments, CSTR2RVAL(":"));
+	ma.type.file.path = RVAL2CSTR(value);
 
 	MAGIC_SYNCHRONIZED(magic_load_internal, &ma);
 	if (ma.status < 0) {

--- a/test/test_magic.rb
+++ b/test/test_magic.rb
@@ -461,7 +461,13 @@ class MagicTest < Test::Unit::TestCase
 
     with_fixtures do
       @magic.load('png-fake.magic')
+      assert_equal(['png-fake.magic'], @magic.paths)
+      assert_match(%r{^Ruby Gem image}, @magic.file(Pathname.new('ruby.png')))
+    end
 
+    with_fixtures do
+      @magic.load('png-fake.magic', 'shell.magic')
+      assert_equal(['png-fake.magic', 'shell.magic'], @magic.paths)
       assert_match(%r{^Ruby Gem image}, @magic.file(Pathname.new('ruby.png')))
     end
   end

